build/root/Makefile.generated_files

```diff
@@ -487,6 +488,108 @@ $(DEFAULTER_GEN):
 	touch $@
 
 #
+# Api linter
```

Comment:
This make logic is so complex that I don't think it provides value (having worked with it in the past).  It's extremely difficult to debug and modify for minor performance gains.  Since in reality most users only see linting in PR jobs, I'd say this is something that should be removed and we should have a simpler invocation that uses normal processing of the packages.  goidl is fast at scanning all packages.

Reply:
I agree on the complexity of makefile. This can be (subjectively) simple scripts. However because the linter needs to work on the same package list as the other generators (specifically OpenAPI generator), do you think it make sense to duplicate that logic in two places? It would be nice (may be in another PR) if we can move all of these together, out of the Makefile into a bash script.